### PR TITLE
Type fixes 2

### PR DESCRIPTION
**Pull request by fbs**
Follow up of #1159

### CODE REVIEW

File: src/ast/codegen_llvm.cpp [hunk 1]
```diff
@@ -1313,7 +1313,9 @@ void CodegenLLVM::visit(AssignMapStatement &assignment)
   Value *val, *expr;
   expr = expr_;
   AllocaInst *key = getMapKey(map);
-  if (map.type.type == Type::string || assignment.expr->type.type == Type::inet)
+  if (map.type.type == Type::string ||
+      assignment.expr->type.type == Type::inet ||
+      assignment.expr->type.type == Type::usym)
```
Comment (danobi) [inline src/ast/codegen_llvm.cpp]: Just to double check: this is ok b/c the usym is already on the stack, right?

**Comment by danobi** [comment]
Looks like you have some unused variable warnings (`assert`s compiled out in release mode) and LLVM api changes to ifdef over, though

File: src/ast/codegen_llvm.cpp (hunk 1 above)
Comment (fbs) [reply src/ast/codegen_llvm.cpp]: Yes.

Found a different bug though  `:f { @=usym(arg0); @a=@ }'` results in a segv. I've added a commit to fix it :)

**Comment by fbs** [comment]
@danobi  I've (hopefully) fixed the tests and fixed a copy bug around usym. Can you take another look?

**Comment by danobi** [comment]
LGTM

**Comment by fbs** [comment]
I'll merge this monday if there are no further comments.

**Pull request merged**